control-plane-operator/controllers/hostedcontrolplane/cvo/params.go

```diff
@@ -33,6 +33,9 @@ func NewCVOParams(hcp *hyperv1.HostedControlPlane, releaseImageProvider *imagepr
 		ClusterID:               hcp.Spec.ClusterID,
 		PlatformType:            hcp.Spec.Platform.Type,
 	}
+	p.DeploymentConfig.AdditionalLabels = map[string]string{
+		config.NeedMetricsServerAccessLabel: "true",
```

Comment:
This isn't currently conditional based on `EnableCVOManagementClusterMetricsAccess`, but the consuming network policy managed in `reconcileNetworkPolicies` is conditional.  Doesn't seem like there's much harm in setting this label in the absence of consumers, but might be cleaner if there's a way to plumb `EnableCVOManagementClusterMetricsAccess` down to this level so you can only set this label when it's required.

Reply:
Nice catch! Let's keep the labels clean, I'll fix this.